ext/io/lib.rs

```diff
@@ -1002,6 +1008,59 @@ impl crate::fs::File for StdFileResourceInner {
   }
 }
 
+pub struct ReadCancelResource(Rc<CancelHandle>);
+
+impl Resource for ReadCancelResource {
+  fn name(&self) -> Cow<str> {
+    "readCancel".into()
+  }
+
+  fn close(self: Rc<Self>) {
+    self.0.cancel();
+  }
+}
+
+#[op2(fast)]
+#[smi]
+pub fn op_read_cancel_handle(state: &mut OpState) -> u32 {
+  state
+    .resource_table
+    .add(ReadCancelResource(CancelHandle::new_rc()))
+}
+
+#[op2(async)]
+pub async fn op_read_cancel(
```

Comment:
Nitpick: at first glance in JS I thought this means "to cancel a read" while in fact it's a "read with maybe cancel". Can you rename this op to `op_read_with_cancel_handle`?